Add docstrings for functions

jkbest2 opened the pull request:
Let me know if I missed anything. I tried to follow [BlueStyle](https://github.com/invenia/BlueStyle#documentation). I can try to write docstrings for the rest of the functions next time I have a chance.

jkbest2 committed:
Add docstrings for precision matrix functions

ElOceanografo commented on src/spde_precision_matrices.jl:
```diff
@@ -1,4 +1,16 @@
+
+# Arguments
+- `mesh::TriMesh`: a mesh from TriangleMesh.jl
+- `κ::Real`: Matérn smoothness parameter
```
This should be "Matérn correlation decay rate parameter"

jkbest2 replied:
Fixed below

ElOceanografo reviewed (changes requested):
Looks good, just one comment.  Thanks for writing these!

jkbest2 committed:
Split calculate_κ and calculate_τ docstrings

jkbest2 committed:
Add docstrings to mesh utility functions

jkbest2 committed:
Fix κ description in `component_matrices`

jkbest2 commented:
Added docstrings for the mesh utility functions and changed the description of κ as requested. I think I got the mesh function descriptions correct, but let me know if there are any more changes.

jkbest2 committed:
Use @doc raw"" to allow math in docstrings

ElOceanografo commented on src/mesh_utils.jl:
```diff
@@ -13,7 +13,20 @@
+"""
+    make_border_points(points, n, expansion, k = size(points, 2)) -> Matrix
 
+Determine points along the edge of the domain.
```
```suggestion
Create a set of border points around the hull of a domain.
```

ElOceanografo commented on src/mesh_utils.jl:
```diff
@@ -13,7 +13,20 @@
+Determine points along the edge of the domain.
+
+# Arguments
+- `points`: points to form a hull around
```
```suggestion
- `points`: points to form a hull around, as a matrix with a point in each column
```

ElOceanografo commented on src/mesh_utils.jl:
```diff
@@ -13,7 +13,20 @@
+- `points`: points to form a hull around
+- `n`: number of edge points
+- `expansion`: factor to extend the hull beyond the provided points
+- `k`: number of neighbors to use when determining hull; larger is smoother
```
```suggestion
- `k`: number of neighbors to use when determining hull. Default is all points, giving a convex hull; can be decreased to make the hull concave.
```

ElOceanografo commented on src/mesh_utils.jl:
```diff
@@ -38,11 +51,34 @@ function make_border_points(points, n, expansion, k=size(points, 2))
+"""
+    observation_matrix(mesh::TriMesh, points::AbstractMatrix) -> SparseMatricCSC
+
+Calculate the weights to interpolate from nodes to observation locations.
```
```suggestion
Create matrix of weights to linearly interpolate from nodes to observation locations.
```

ElOceanografo reviewed (changes requested):
Looking good, just some minor wording edits

jkbest2 committed:
Apply suggestions from code review

Co-authored-by: Sam <oceanographerschoice@gmail.com>

jkbest2 commented:
Thanks for the edits; I applied them. This should be ok to merge now.

ElOceanografo commented:
Looks good, thanks!
